packages/cdktf/lib/iterator.ts

```diff
@@ -0,0 +1,200 @@
+import { TerraformStack } from ".";
+import { ComplexList, ComplexMap } from "./complex-computed-list";
+import { Fn } from "./terraform-functions";
+import { propertyAccess, ref } from "./tfExpression";
+import { Lazy, Token } from "./tokens";
+
+export interface IIterator {
+  /**
+   * @internal used by TerraformResource to set the for_each expression
+   */
+  _getForEachExpression(): any;
+}
+
+export abstract class Iterator implements IIterator {
+  /**
+   * @internal used by TerraformResource to set the for_each expression
+   */
+  abstract _getForEachExpression(): any;
+
+  /**
+   * Creates a new iterator from a list
+   */
+  public static fromList(
+    list: string[] | number[] | ComplexList
+  ): ListIterator {
+    // TODO: this could return different iterators depending on the type of the list
+    // for example it could return a NumberListIterator whose iterator.key would be a number
+    return new ListIterator(list);
+  }
+
+  /**
+   * Creates a new iterator from a map
+   */
+  public static fromMap(
+    map:
+      | ComplexMap
+      | { [key: string]: any }
+      | { [key: string]: string }
+      | { [key: string]: number }
+      | { [key: string]: boolean }
+  ): MapIterator {
+    return new MapIterator(map);
+  }
+
+  /**
+   * @param attribute name of the property to retrieve
+   * @returns the given attribute of the current item iterated over as a string
+   */
+  getString(attribute: string): string {
```

Comment:
Not all `get<Type>` methods are present that we support. One example is `getBoolean`

Reply:
Ah, right

Reply:
added `getBoolean, getAny, getAnyMap`